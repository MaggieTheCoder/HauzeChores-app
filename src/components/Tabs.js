import React, { useState } from "react";
import "../styles/Tabs.css";

const Tabs = () => {
  const [activeIndex, setActiveIndex] = useState(1);
  const handleClick = (index) => {
    setActiveIndex(index);
    console.log("button clicked", index);
  };

  const checkActive = (index, className) =>
    activeIndex === index ? className : "";

  return (
    <>
      <div className="tabs">
        <button
          type="button"
          className={`tab ${checkActive(1, "active")}`}
          onClick={() => handleClick(1)}
        >
          All Tasks{" "}
        </button>
        <button
          type="button"
          className={`tab ${checkActive(2, "active")}`}
          onClick={() => handleClick(2)}
        >
          My Tasks{" "}
        </button>
        <button
          type="button"
          className={`tab ${checkActive(3, "active")}`}
          onClick={() => handleClick(3)}
        >
          My Streak{" "}
        </button>
      </div>
      {activeIndex === 1 && (
        <div className="panels">
          <div className={`panel ${checkActive(1, "active")}`}>
            <p>
              {" "}
              Tab 1 Lorem ipsum dolor sit amet, consectetur adipiscing elit.
              Aenean erat ligula, feugiat at felis vitae, porttitor lacinia
              quam.
            </p>
          </div>
        </div>
      )}

      {activeIndex === 2 && (
        <div className="panels">
          <div className={`panel ${checkActive(1, "active")}`}>
            <p>
              {" "}
              Tab 2 Lorem ipsum dolor sit amet, consectetur adipiscing elit.
              Aenean erat ligula, feugiat at felis vitae, porttitor lacinia
              quam.
            </p>
          </div>
        </div>
      )}

      {activeIndex === 3 && (
        <div className="panels">
          <div className={`panel ${checkActive(1, "active")}`}>
            <p>
              {" "}
              Tab 3 Lorem ipsum dolor sit amet, consectetur adipiscing elit.
              Aenean erat ligula, feugiat at felis vitae, porttitor lacinia
              quam.
            </p>
          </div>
        </div>
      )}
    </>
  );
};

export default Tabs;
