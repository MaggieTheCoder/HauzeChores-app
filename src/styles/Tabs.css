.tabs {
    display: flex;
}

.tab {
    justify-content: center;
    align-items: center;
    width: 100%;
    background-color: lightsteelblue;
    padding: 2em 0;
    border: none;
}

.tab:not(:last-child) {
    border-right: 1px solid steelblue
}

.tab.active {
    background: steelblue;
    font-weight: bold;
}

.tab-content__panel {
    display: none;
    background-color: #eaeaea;
    padding: 10px;
}

.tab-content__panel:active {
    display: block;
}