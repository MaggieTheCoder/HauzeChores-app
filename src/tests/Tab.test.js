import React from "react";
import { render, screen } from "@testing-library/react";
import Tabs from "../components/Tabs";

describe("Tab component", () => {
  const validProps = {
    houseTasks: [],
    setAddedANewTask: () => {},
    userId: 2,
    userTasks: [],
  };
  test("it renders correctly", () => {
    const { asFragment } = render(
      <Tabs
        houseTasks={validProps.houseTasks}
        setAddedANewTask={validProps.setAddedANewTask}
        userId={validProps.userId}
        userTasks={validProps.userTasks}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Tab buttons render correctly", () => {
    render(
      <Tabs
        houseTasks={validProps.houseTasks}
        setAddedANewTask={validProps.setAddedANewTask}
        userId={validProps.userId}
        userTasks={validProps.userTasks}
      />
    );
    const button1 = screen.getByText("All Tasks");
    const button2 = screen.getByText("My Tasks");
    const button3 = screen.getByText("Leaderboard");

    expect(button1).toBeInTheDocument();
    expect(button2).toBeInTheDocument();
    expect(button3).toBeInTheDocument();
  });
});
