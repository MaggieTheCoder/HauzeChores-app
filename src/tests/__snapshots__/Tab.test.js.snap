// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="tabs"
  >
    <button
      class="tab active"
      type="button"
    >
      All Tasks 
    </button>
    <button
      class="tab "
      type="button"
    >
      My Tasks 
    </button>
    <button
      class="tab "
      type="button"
    >
      Leaderboard 
    </button>
  </div>
  <div
    class="panels"
  >
    <div
      class="panel active"
    />
  </div>
</DocumentFragment>
`;
